react.js

```diff
@@ -0,0 +1,58 @@
+"use strict";
+
+const baseConfig = require(".");
+
+const eslintReactRecommendedRules = {
+  "react/display-name": 2,
+  "react/jsx-key": 2,
+  "react/jsx-no-comment-textnodes": 2,
+  "react/jsx-no-duplicate-props": 2,
+  "react/jsx-no-target-blank": 2,
+  "react/jsx-no-undef": 2,
+  "react/jsx-uses-react": 2,
+  "react/jsx-uses-vars": 2,
+  "react/no-children-prop": 2,
+  "react/no-danger-with-children": 2,
+  "react/no-deprecated": 2,
+  "react/no-direct-mutation-state": 2,
+  "react/no-find-dom-node": 2,
+  "react/no-is-mounted": 2,
+  "react/no-render-return-value": 2,
+  "react/no-string-refs": 2,
+  "react/no-unescaped-entities": 2,
+  "react/no-unknown-property": 2,
+  "react/no-unsafe": 0,
+  "react/prop-types": 2,
+  "react/react-in-jsx-scope": 2,
+  "react/require-render-return": 2,
+};
+
+module.exports = {
+  ...baseConfig,
+  rules: {
+    ...baseConfig.rules,
+    ...eslintReactRecommendedRules,
+    "react/prop-types": 0,
+    "react/react-in-jsx-scope": 2,
+    "react/jsx-first-prop-new-line": [ 2, "multiline" ],
+    "react/jsx-max-props-per-line": [ 2, { maximum: { single: 3, multi: 1 } } ],
+    "react/jsx-closing-bracket-location": [ 2, "tag-aligned" ],
+    "react/jsx-uses-react": 2,
+    "react/destructuring-assignment": [ 2, "always" ],
+    "react/prefer-stateless-function": [ 2, { ignorePureComponents: false } ],
+    "react/function-component-definition": [ 1, { namedComponents: "arrow-function" } ],
```

Comment:
föredrar function expressions

```js
function Component { … }
```

istället för

```js
const Component = () => { … }
```

Reply:
intressant läsning kanske? => https://stackoverflow.com/questions/49306148/why-is-arrow-syntax-preferred-over-function-declaration-for-functional-react-com

Reply:
> föredrar function expressions

"function declaration" hette det visst.

Rent praktiskt tycker jag det är enklare att skriva: 
"function" + <namn> + `shift+(8, 9)` + `space` + `shift + alt + (8+9)`
vs
"const" + <namn> + `shift + 0` + `space` + `shift + (8, 9)` + `space` + `shift + 0 + pil` + `space` + `shift + alt + (8+9)`.

Mer optimerat för autocomplete också om man använder det: "fun"+`tab`. värdet av const kan ju vara vad som helst.

Reply:
func declaration för named och arrow unamed.

Reply:
@WillB01, den enda (bland topp-svar/-kommentarer) som förespråkar const-funktioner säger bara "It looks a lot cooler".

De som talar emot hänvisar till svårigheter att läsa koden, `this`-hantering och Reacts officiella dokumentation. 🙂